integration_test.go

```diff
@@ -1001,72 +1001,113 @@ test: 1
 	}
 }
 
-func TestGRPCStreamWithStreamIntervalOption(t *testing.T) {
-	t.Run("client stream", func(t *testing.T) {
-		addr, server := setupGRPCServer(t, &simpleService{
+func TestGRPCStreamWithBoundedExecutionTime(t *testing.T) {
+	tests := []struct {
+		desc         string
+		opts         Options
+		wantExecTime time.Duration
+
+		returnOutput  []simple.Foo
+		expectedInput []simple.Foo
+	}{
+		{
+			desc: "client side streaming with stream interval option",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ClientStream",
+					Timeout:           timeMillisFlag(time.Second * 5),
+					RequestJSON:       `{"test":1}{"test":2}{"test":-1}`,
+					StreamRequestOptions: StreamRequestOptions{
+						Interval: timeMillisFlag(time.Millisecond * 1000),
+					},
+				},
+			},
 			returnOutput:  []simple.Foo{{Test: 4}},
 			expectedInput: []simple.Foo{{Test: 1}, {Test: 2}, {Test: -1}},
-		})
-		defer server.Stop()
-
-		opts := Options{
-			ROpts: RequestOptions{
-				FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
-				Procedure:         "Bar/ClientStream",
-				Timeout:           timeMillisFlag(time.Second * 5),
-				RequestJSON:       `{"test":1}{"test":2}{"test":-1}`,
-				StreamRequestOptions: StreamRequestOptions{
-					Interval: timeMillisFlag(time.Millisecond * 1000),
+			// Test must run for more than 2 seconds as there are three requests and
+			// it must take 2 seconds totally between consecutive messages.
+			wantExecTime: time.Second * 2,
```

Comment:
here and below, since tests will wait for the timeout, should we use shorter values?

(ideally we wouldn't have tests sleeping at all, but I think that'll need a larger refactor, so something to note for future)

Reply:
I have reduced the time to ~100ms for all the test cases. I agree, I'll try to refactor this with a mock clock setup.